pwnlib/gdb.py

```diff
@@ -751,23 +751,35 @@ def attach(target, gdbscript = '', exe = None, gdb_args = None, ssh = None, sysr
         b'Hello from bash\n'
 
         Using GDB Python API:
-
-        >>> if six.PY3:
-        ...     io = process('bash')
-        ...     # Attach a debugger
-        ...     pid, io_gdb = gdb.attach(io, api=True)
-        ...     # Force the program to write something it normally wouldn't
-        ...     io_gdb.execute('call puts("Hello from process debugger!")')
-        ...     # Resume the program
-        ...     io_gdb.continue_nowait()
-        ...     # Observe the forced line
-        ...     s = io.recvline()
-        ...     assert s == b'Hello from process debugger!\n', s
-        ...     # Interact with the program in a regular way
-        ...     io.sendline('echo Hello from bash && exit')
-        ...     # Observe the results
-        ...     s = io.recvall()
-        ...     assert s == b'Hello from bash\n', s
+        
+        .. doctest
+           :skipif: six.PY2
+
+            >>> io = process('bash')
+            >>> # Attach a debugger
+            >>> pid, io_gdb = gdb.attach(io, api=True)
+            
+            Force the program to write something it normally wouldn't
+            
+            >>> io_gdb.execute('call puts("Hello from process debugger!")')
+            
+            Resume the program
+            
+            >>> io_gdb.continue_nowait()
+            
+            Observe the forced line
+            
+            >>> s = io.recvline()
+            >>> assert s == b'Hello from process debugger!\n', s
+            
+            Interact with the program in a regular way
+            
+            >>> io.sendline('echo Hello from bash && exit')
+            
+            Observe the results
+            
+            >>> s = io.recvall()
+            >>> assert s == b'Hello from bash\n', s
```

Comment:
Maybe shorter BTW?
```suggestion
            >>> io.recvall()
            b'Hello from bash\n'
```

Reply:
I don't want to touch any of the GDB tests (I think this proves I did by accident) because the tests don't run in my Docker container world.  This is an easy and no-chance-of-failing-catastrophically-change so I'll make it.